greenlight/lib/tests/test_tabs.py

```diff
@@ -47,9 +47,7 @@ def tearDown(self):
             gBrowser.removeAllTabsBut(gBrowser.tabs[0]);
         """)
 
-        # Needs to be set by the harness itself
-        self.prefs.restore_pref('browser.tabs.warnOnClose')
-        self.prefs.restore_pref('browser.tabs.warnOnCloseOtherTabs')
+        self.wait_for_condition(lambda _: len(self.browser.tabbar.tabs) == 1)
```

Comment:
This fails the assertion I added, because removeAllTabsBut is (apparently) asynchronous, but we don't wait. Removing the restore_pref calls is unrelated, but we don't need them now with your restore_all method.